core/amber/src/main/scala/edu/uci/ics/amber/engine/common/statetransition/WorkerStateManager.scala

```diff
@@ -0,0 +1,34 @@
+package edu.uci.ics.amber.engine.common.statetransition
+
+import edu.uci.ics.amber.engine.common.statetransition.StateManager.IntermediateState
+import edu.uci.ics.amber.engine.common.statetransition.WorkerStateManager._
+
+object WorkerStateManager {
+  sealed abstract class WorkerState extends Product with Serializable
```

Comment:
why does WorkerState extend Product?

Reply:
https://contributors.scala-lang.org/t/could-adts-extend-product-with-serializable-in-dotty/3304. This article shows we sometimes will have issues on type inference if we don't extend Product. And this is a common practice. I will add a comment.